### ISSUE TITLE
Unable to install zope-interface with python 3.9 for v4.7.1 zope-interface

### ISSUE THREAD

**Issue by rlam3** [zopefoundation/zope.interface#246]
### What I did:

Tried installing humming bot from source and it looks like it doesn't support python 3.9
https://github.com/CoinAlpha/hummingbot-docs/blob/9dbe2cab183de55cfca7c9316a11abcb8b21f89f/content/installation/mac.mdx

### What actually happened:

```
Collecting zope-interface==4.7.1
  Using cached zope.interface-4.7.1.tar.gz (151 kB)

Pip subprocess error:
  Running command git clone -q https://github.com/CoinAlpha/python-signalr-client.git /private/var/folders/th/c91sj7ss2m173ylmh8xf7g_m0000gn/T/pip-req-build-9hq7zeh3
    ERROR: Command errored out with exit status 1:
     command: /Users/XXXXXX/opt/anaconda3/envs/hummingbot/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/th/c91sj7ss2m173ylmh8xf7g_m0000gn/T/pip-install-g3wm0rm6/zope-interface_ddc133df7b2141cc81aee1787b3f7657/setup.py'"'"'; __file__='"'"'/private/var/folders/th/c91sj7ss2m173ylmh8xf7g_m0000gn/T/pip-install-g3wm0rm6/zope-interface_ddc133df7b2141cc81aee1787b3f7657/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/th/c91sj7ss2m173ylmh8xf7g_m0000gn/T/pip-pip-egg-info-lrgs6tkz
         cwd: /private/var/folders/th/c91sj7ss2m173ylmh8xf7g_m0000gn/T/pip-install-g3wm0rm6/zope-interface_ddc133df7b2141cc81aee1787b3f7657/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/th/c91sj7ss2m173ylmh8xf7g_m0000gn/T/pip-install-g3wm0rm6/zope-interface_ddc133df7b2141cc81aee1787b3f7657/setup.py", line 26, in <module>
        from setuptools import setup, Extension, Feature
    ImportError: cannot import name 'Feature' from 'setuptools' (/Users/XXXXXX/opt/anaconda3/envs/hummingbot/lib/python3.8/site-packages/setuptools/__init__.py)
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/c3/05/bf3130eb799548882ce61b7c3d2dbc5d4d5cc6e821efa8786c5273d56844/zope.interface-4.7.1.tar.gz#sha256=4bb937e998be9d5e345f486693e477ba79e4344674484001a0b646be1d530487 (from https://pypi.org/simple/zope-interface/) (requires-python:>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement zope-interface==4.7.1 (from versions: 3.3.0b1, 3.3.0b2, 3.3.0, 3.3.0.1, 3.4.0b1, 3.4.0, 3.4.1, 3.5.0, 3.5.1, 3.5.2, 3.5.3, 3.6.0, 3.6.1, 3.6.2, 3.6.3, 3.6.4, 3.6.5, 3.6.6, 3.6.7, 3.6.8, 3.7.0, 3.8.0, 4.0.0, 4.0.1, 4.0.2, 4.0.3, 4.0.4, 4.0.5, 4.1.0, 4.1.1, 4.1.2, 4.1.3, 4.2.0, 4.3.0, 4.3.1, 4.3.2, 4.3.3, 4.4.0, 4.4.1, 4.4.2, 4.4.3, 4.5.0, 4.6.0, 4.7.0, 4.7.1, 4.7.2, 5.0.0, 5.0.1, 5.0.2, 5.1.0, 5.1.1, 5.1.2, 5.2.0, 5.3.0a1, 5.3.0, 5.4.0)
ERROR: No matching distribution found for zope-interface==4.7.1

failed

CondaEnvException: Pip failed
```

### What version of Python and Zope/Addons I am using:

python: 3.9
zope: 4.7.1

**Comment by rlam3** [comment]
https://github.com/zopefoundation/zope.interface/pull/247

**Comment by jugmac00** [comment]
Hi, @rlam3 - `zope.interace` 4.7.1 did not support Python 3.9 yet - you need to install a newer version, or tell the maintainers of the project which uses `zope.interface` not to pin such an old version, at least not when they want to support Python 3.9.

**Comment by jensens** [comment]
If we want to express this explicit, we could add a `<3.9` to the list here https://github.com/zopefoundation/zope.interface/blob/30815e97af70465c02a72fb2a932cccde66f770e/setup.py#L135-L142 This way users of Python 3.9 and of upcoming Python versions are prevented running into this trap. Adding each upcoming version separate is pointless.

**Comment by icemac** [comment]
@rlam3 This looks more like a problem with the version of `setuptools` you are using – than `zope.interface.` As the version number is not in the traceback, I'd like you to add you this information in a comment.

**Comment by icemac** [comment]
@rlam3 `zope.interface` 4.7.1 was released at 2019-11-11. So you probably need a `setuptools` version from around that time. See #30 where the problem was also discussed and fixed in 4.7.2. You should probably report to the Hummingbot developers that their documentation resp. versions used are outdated.

**Comment by icemac** [comment]
Closed due to timeout. Feel free to re-open if there are any news.
